src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -115,33 +115,108 @@ public static void Analyze(CSharpCompilation compilation, Symbol member, BoundNo
 
         protected override void Normalize(ref LocalState state)
         {
+            int oldNext = state.Capacity;
             state.EnsureCapacity(nextVariableSlot);
+            for (int i = oldNext; i < nextVariableSlot; i++)
+            {
+                var id = variableBySlot[i];
+                // Initialize non-struct members with the declared nullability.
+                // (Struct members are handled by InheritNullableStateOfTrackableStruct.)
+                bool? isNotNull = null;
+                int slot = id.ContainingSlot;
+                if (slot > 0 && variableBySlot[slot].Symbol.GetTypeOrReturnType().TypeKind != TypeKind.Struct)
+                {
+                    isNotNull = GetMemberIsNotNull(id.Symbol);
+                }
+                state[i] = isNotNull;
+            }
         }
 
-        protected override bool TryGetReceiverAndMember(BoundExpression expr, out BoundExpression receiver, out Symbol member)
+        private void Assign(ref LocalState state, int slot, bool? isNotNull)
```

Comment:
>private void Assign(ref LocalState state, int slot, bool? isNotNull) [](start = 8, length = 68)

It feels like we don't need to introduce new tracking system, It should be sufficient to extend what we do for anonymous types. Also we do want to copy state on assignment rather than reset it.

Reply:
Changed to use the existing mechanism.